content/docs/reference-react-dom-server.md

```diff
@@ -17,93 +17,89 @@ var ReactDOMServer = require('react-dom/server');
+The following methods can be used in both the server and browser environments:
 
-If you plan to use React on the client to make the markup interactive, do not use this method. Instead, use [`renderToString`](#rendertostring) on the server and [`ReactDOM.hydrateRoot()`](/docs/react-dom-client.html#hydrateroot) on the client.
+- [`renderToString()`](#rendertostring)
+- [`renderToStaticMarkup()`](#rendertostaticmarkup)
 
-* * *
+## Reference {#reference}
 
 ### `renderToPipeableStream()` {#rendertopipeablestream}
 
 ```javascript
 ReactDOMServer.renderToPipeableStream(element, options)
 ```
 
-Render a React element to its initial HTML. Returns a [Control object](https://github.com/facebook/react/blob/3f8990898309c61c817fbf663f5221d9a00d0eaa/packages/react-dom/src/server/ReactDOMFizzServerNode.js#L49-L54) that allows you to pipe the output or abort the request. Fully supports Suspense and streaming of HTML with "delayed" content blocks "popping in" later through javascript execution. [Read more](https://github.com/reactwg/react-18/discussions/37)
+Render a React element to its initial HTML. Returns a stream with a `pipe(res)` method to pipe the output and `abort()` to abort the request. Fully supports Suspense and streaming of HTML with "delayed" content blocks "popping in" via inline `<script>` tags later. [Read more](https://github.com/reactwg/react-18/discussions/37)
 
 If you call [`ReactDOM.hydrateRoot()`](/docs/react-dom-client.html#hydrateroot) on a node that already has this server-rendered markup, React will preserve it and only attach event handlers, allowing you to have a very performant first-load experience.
 
-> Note:
->
-> This is a Node.js specific API and modern server environments should use renderToReadableStream instead.
->
-
-```
-const {pipe, abort} = renderToPipeableStream(
+```javascript
+let didError = false;
+const stream = renderToPipeableStream(
   <App />,
   {
-    onAllReady() {
-      res.statusCode = 200;
+    onShellReady() {
+      // The content above all Suspense boundaries is ready.
+      // If something errored before we started streaming, we set the error code appropriately.
+      res.statusCode = didError ? 500 : 200;
       res.setHeader('Content-type', 'text/html');
-      pipe(res);
+      stream.pipe(res);
     },
-    onShellError(x) {
+    onShellError(error) {
+      // Something errored before we could complete the shell so we emit an alternative shell.
       res.statusCode = 500;
       res.send(
         '<!doctype html><p>Loading...</p><script src="clientrender.js"></script>'
       );
-    }
+    },
+    onAllReady() {
+      // If you don't want streaming, use this instead of onShellReady.
+      // This will fire after the entire page content is ready.
+      // You can use this for crawlers (SEO) or static generation.
```

Comment:
I'm not sure we should mention SEO here because these days, it's not actually mainly for SEO bots since they support JS. It's actually mainly for other types of dumb bots like social network's OpenGraph parsers etc.